### PR TITLE
DCSPII-495: Update country I am applying from in the filter - use new…

### PR DESCRIPTION
**Pull request by steven-hughes**
* Update countries data model to match v1 api response

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -62,15 +64,31 @@ class CountriesCachedModel {
...
-            country => country.id !== 'UK'
+            country => country.countryCode !== 'GB'
+        );
+        this._residenceCountries = _.filter(
+            countries,
+            country => country.addressCountryFlag === true
+        );
+        this._overseasResidenceCountries = _.filter(
+            this._residenceCountries,
+            country => country.addressCountryFlag !== 'GB'
```
Comment (HughePaul) [inline lib/index.js]: Should this be checking both the countryCode and the flag?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hmpo-countries-lib",
-  "version": "1.4.0",
+  "version": "2.0.0",
```
Comment (HughePaul) [inline package.json]: Don't update the version until it is published

File: test/lib/spec.index.js [hunk 3]
```diff
@@ -334,8 +373,8 @@ describe('CountriesCachedModel', () => {
 
         it('should return null if no policy data is found', () => {
             instance._indexCountries();
-            delete instance._policiesById['UK'];
-            expect(instance.getCountryDataById('UK')).to.equal(null);
+            delete instance._policiesById['GB'];
```
Comment (HughePaul) [inline test/lib/spec.index.js]: UK policy is still indexed by UK; there should be a test for a non UK country

File: test/lib/spec.index.js [hunk 4]
```diff
@@ -216,32 +246,40 @@ describe('CountriesCachedModel', () => {
...
+                { addressCountryFlag: false, countryCode: 'BA', countryNameSlug: 'bar' },
+                { addressCountryFlag: null, countryCode: 'NA', countryNameSlug: 'narnia' }
+            ]);
+        });
+
+        it('should set _residenceCountries to all countries where addressCountryFlag is true', () => {
+            instance._indexCountries();
+            instance._residenceCountries.should.deep.equal([
+                { addressCountryFlag: true, countryCode: 'GB', countryNameSlug: 'united-kingdom' },
+                { addressCountryFlag: true, countryCode: 'FO', countryNameSlug: 'foo' }
```
Comment (HughePaul) [inline test/lib/spec.index.js]: There should also be a test for resident overseas countries indexing

Comment (steven-hughes) [reply test/lib/spec.index.js]: done

**Pull request merged**